### DISCUSSION TITLE
ESM only for node-wifi major release 3.0

### DISCUSSION THREAD

**Discussion by friedrith** [friedrith/node-wifi#170] [Ideas]
A maintainer proposed to switch to ES Modules for next major release. Since I also think it is a good to promote changes when they are going into the good direction, I tend to switch to ES Modules too.

## What are ES Modules 

ES Modules is the ECMAScript standard for working with modules. For more details, you can read [this article](https://flaviocopes.com/es-modules/).

## What will be the impacts?

In your code, it will mean replacing `const nodeWifi = require('node-wifi')` by `import nodeWifi from 'node-wifi'` or maybe `import { scanWifi } from 'node-wifi'`.

If you are using the library `node-wifi` in a commonJs code or library, you will need to change your code to use ES Modules.

If you are already using the ES Module for your own code, you won't have to change anything.

## Why (only) ES Modules?

Node.js was previously the standard CommonJS. But in opposite of CommonJS, ES Modules are used by all other js environments like browsers.  

Node.js supports ES Modules since [version 12.x[(https://nodejs.org/docs/latest-v12.x/api/esm.html). Previous Node.js version which was not supporting ES Modules (Version 10.x) is [no more supported by Node.js community](https://endoflife.date/nodejs) (even security support). So I think it is the good timing to consider Node.js version 10.x as deprecated and so everybody should be able to use ES Modules.

An option would have been to support and export ES Modules in addition of current CommonJS modules. But so far several reasons incitate to keep only ES Modules:
- less standards to support means more resources for useful features and bug fixes 👍 
- lighter repo and continuous delivery: always good for environment 🌍 
- to push the community to adopt ES Modules (which are far better IMHO)

For more details about choices, you can read [this thread](https://github.com/sindresorhus/meta/discussions/15) from a famous developer and packages maintainer.

**What is your own opinion? Do not hesitate to discuss and give your point of view.**

PS: thanks to @jimmywarting for his proposal

**Comment by AWAIS-514** [comment]
It not work with Os 64 bit?

**Comment by wobsoriano** [comment]
We can use https://tsup.egoist.sh/ to output this in both esm and cjs. I can do a PR if you're interested

**Comment by jitcoder** [comment]
imo: ESM is the way to go. switched to ESM a year ago and haven't looked back.

**Reply by tunafysh** [reply]
Same here honestly. Ever since started using react, ESM was way better.
